### PR TITLE
Use `fbgemm` for quantize/dequantize ops

### PR DESCRIPTION
**Pull request by z-a-f**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#19500 Use `fbgemm` for quantize/dequantize ops**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D15014561/)

Changes the `quantize_linear` and `dequantize` to `fbgemm`-based implementation.

Differential Revision: [D15014561](https://our.internmc.facebook.com/intern/diff/D15014561/)

### CODE REVIEW

File: aten/src/ATen/quantized/Quantizer.cpp [hunk 1]
```diff
@@ -76,12 +86,23 @@ QTensor PerTensorAffineQuantizer::quantize(RealTensor tensor) {
       sizes,
       tensor.options().dtype(at::kQInt8),
       intrusive_from_this());
-  auto qvd = qv.data<qint8>();
-  tensor.contiguous();
```
Comment (ezyang) [inline aten/src/ATen/quantized/Quantizer.cpp]: Nice catch. Would be good to have a test for this case.

**Comment by ezyang** [comment]
Change looks fine to me. Resigning in favor of other reviewers.

**Comment by ezyang** [comment]
Is FBGEMM codepath being exercised in CI? What about the non-FBGEMM codepath?

**Comment by zafartahirov** [comment]
> Is FBGEMM codepath being exercised in CI? What about the non-FBGEMM codepath?

I am not sure if CI sets the `USE_FBGEMM` flag. Ideally it should -- in case there are any OSS tests that might fail.

**Pull request closed without merging**